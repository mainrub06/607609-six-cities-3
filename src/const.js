export const STAR_PARAMS = {
  MIN_VALUE: 0,
  MAX_VALUE: 5,
  STAR_STEP: 20
};

export const LINKS = {
  INDEX: `/`,
  OFFER_DETAIL: `/offer/`,
  LOGIN: `/login`,
  FAVORITES: `/favorites`
};

export const OFFERS_SORT_ITEMS = [`Popular`, `Price: low to high`, `Price: high to low`, `Top rated first`];

export const CITIES = [
  `Paris`,
  `Cologne`,
  `Brussels`,
  `Amsterdam`,
  `Hamburg`,
  `Dusseldorf`
];

export const MAX_PHOTOS_OFFER_DETAIL = 6;

export const AUTHORIZATION_STATUS = {
  AUTH: `AUTH`,
  NO_AUTH: `NO_AUTH`
};

export const FAVORITE_REQUESTS = {
  ADD: `1`,
  DELETE: `0`
};

export const REQUEST_STATUS = {
  NO_AUTH: 401,
  OK: 200
};

export const OFFERS_CSS_CLASSES = {
  MAIN: {
    LIST: `cities__places-list`,
    ITEM: `cities__place-card`,
    IMAGE_WRAPPER: `cities__image-wrapper`,
    ITEM_INFO: ``,
    IMAGE_SIZE: {
      WIDTH: 260,
      HEIGHT: 200
    }
  },
  OFFER_DETAIL: {
    LIST: `near-places__list`,
    ITEM: `near-places__card`,
    IMAGE_WRAPPER: `near-places__image-wrapper`,
    ITEM_INFO: ``,
    IMAGE_SIZE: {
      WIDTH: 260,
      HEIGHT: 200
    }
  },
  FAVORITE: {
    LIST: `favorites__places`,
    ITEM: `favorites__card`,
    IMAGE_WRAPPER: `favorites__image-wrapper`,
    ITEM_INFO: `favorites__card-info`,
    IMAGE_SIZE: {
      WIDTH: 150,
      HEIGHT: 110
    }
  }
};

export const DETAIL_PAGE_PARAMS = {
  NEAR_OFFERS_MAX: 3
};

export const FORM_PARAMS = {
  STARS: [`perfect`, `good`, `not bad`, `badly`, `terribly`],
  MIN_TEXTAREA_VALUE: 50,
  MAX_TEXTAREA_VALUE: 300,
  MIN_STARS_CHECKED: 1
};

export const REVIEWS_PARAMS = {
  MAX_REVIEWS: 10
};
