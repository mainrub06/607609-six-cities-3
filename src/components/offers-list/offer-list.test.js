import React from "react";
import renderer from "react-test-renderer";
import OfferList from "./offers-list.jsx";
import PLACE_MOCK from "../../mock/offers";
import {OFFERS_CSS_CLASSES} from "../../const";
import {MemoryRouter} from "react-router-dom";

it(`Render snapshot <OfferList/>`, () => {
  const tree = renderer
    .create(
        <MemoryRouter>
          <OfferList offersCssClasses = {OFFERS_CSS_CLASSES.MAIN} handleClickFavoriteButton = {() => {}} handleOfferHover = {() => {}} onOfferClick={() => {}} dataCards={PLACE_MOCK} />
        </MemoryRouter>
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});
