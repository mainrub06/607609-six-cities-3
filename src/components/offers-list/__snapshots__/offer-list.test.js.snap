// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render snapshot <OfferList/> 1`] = `
<div
  className="cities__places-list places__list"
>
  <article
    className="cities__place-card place-card"
    onMouseLeave={[Function]}
    onMouseOver={[Function]}
  >
    <div
      className="place-card__mark"
    >
      <span>
        Premium
      </span>
    </div>
    <div
      className="cities__image-wrapper place-card__image-wrapper"
    >
      <a
        href="#"
      >
        <img
          alt="Place image"
          className="place-card__image"
          height="200"
          src="img/apartment-01.jpg"
          width="260"
        />
      </a>
    </div>
    <div
      className=" place-card__info"
    >
      <div
        className="place-card__price-wrapper"
      >
        <div
          className="place-card__price"
        >
          <b
            className="place-card__price-value"
          >
            €
            100
          </b>
          <span
            className="place-card__price-text"
          >
            / night
          </span>
        </div>
        <button
          className=" place-card__bookmark-button button"
          onClick={[Function]}
          type="button"
        >
          <svg
            className="place-card__bookmark-icon"
            height="19"
            width="18"
          >
            <use
              xlinkHref="#icon-bookmark"
            />
          </svg>
          <span
            className="visually-hidden"
          >
            To bookmarks
          </span>
        </button>
      </div>
      <div
        className="place-card__rating rating"
      >
        <div
          className="place-card__stars rating__stars"
        >
          <span
            style={
              Object {
                "width": "60%",
              }
            }
          />
          <span
            className="visually-hidden"
          >
            Rating
          </span>
        </div>
      </div>
      <h2
        className="place-card__name"
        onClick={[Function]}
      >
        <a
          href="/offer/0221552"
          onClick={[Function]}
        >
          Beautiful & luxurious apartment at great location
        </a>
      </h2>
      <p
        className="place-card__type"
      >
        room
      </p>
    </div>
  </article>
  <article
    className="cities__place-card place-card"
    onMouseLeave={[Function]}
    onMouseOver={[Function]}
  >
    <div
      className="place-card__mark"
    >
      <span>
        Premium
      </span>
    </div>
    <div
      className="cities__image-wrapper place-card__image-wrapper"
    >
      <a
        href="#"
      >
        <img
          alt="Place image"
          className="place-card__image"
          height="200"
          src="img/apartment-02.jpg"
          width="260"
        />
      </a>
    </div>
    <div
      className=" place-card__info"
    >
      <div
        className="place-card__price-wrapper"
      >
        <div
          className="place-card__price"
        >
          <b
            className="place-card__price-value"
          >
            €
            120
          </b>
          <span
            className="place-card__price-text"
          >
            / night
          </span>
        </div>
        <button
          className=" place-card__bookmark-button button"
          onClick={[Function]}
          type="button"
        >
          <svg
            className="place-card__bookmark-icon"
            height="19"
            width="18"
          >
            <use
              xlinkHref="#icon-bookmark"
            />
          </svg>
          <span
            className="visually-hidden"
          >
            To bookmarks
          </span>
        </button>
      </div>
      <div
        className="place-card__rating rating"
      >
        <div
          className="place-card__stars rating__stars"
        >
          <span
            style={
              Object {
                "width": "80%",
              }
            }
          />
          <span
            className="visually-hidden"
          >
            Rating
          </span>
        </div>
      </div>
      <h2
        className="place-card__name"
        onClick={[Function]}
      >
        <a
          href="/offer/5464564"
          onClick={[Function]}
        >
          Wood and stone place
        </a>
      </h2>
      <p
        className="place-card__type"
      >
        apartment
      </p>
    </div>
  </article>
  <article
    className="cities__place-card place-card"
    onMouseLeave={[Function]}
    onMouseOver={[Function]}
  >
    <div
      className="place-card__mark"
    >
      <span>
        Premium
      </span>
    </div>
    <div
      className="cities__image-wrapper place-card__image-wrapper"
    >
      <a
        href="#"
      >
        <img
          alt="Place image"
          className="place-card__image"
          height="200"
          src="img/apartment-03.jpg"
          width="260"
        />
      </a>
    </div>
    <div
      className=" place-card__info"
    >
      <div
        className="place-card__price-wrapper"
      >
        <div
          className="place-card__price"
        >
          <b
            className="place-card__price-value"
          >
            €
            210
          </b>
          <span
            className="place-card__price-text"
          >
            / night
          </span>
        </div>
        <button
          className=" place-card__bookmark-button button"
          onClick={[Function]}
          type="button"
        >
          <svg
            className="place-card__bookmark-icon"
            height="19"
            width="18"
          >
            <use
              xlinkHref="#icon-bookmark"
            />
          </svg>
          <span
            className="visually-hidden"
          >
            To bookmarks
          </span>
        </button>
      </div>
      <div
        className="place-card__rating rating"
      >
        <div
          className="place-card__stars rating__stars"
        >
          <span
            style={
              Object {
                "width": "80%",
              }
            }
          />
          <span
            className="visually-hidden"
          >
            Rating
          </span>
        </div>
      </div>
      <h2
        className="place-card__name"
        onClick={[Function]}
      >
        <a
          href="/offer/4589512"
          onClick={[Function]}
        >
          Radisson SAS
        </a>
      </h2>
      <p
        className="place-card__type"
      >
        hotel
      </p>
    </div>
  </article>
  <article
    className="cities__place-card place-card"
    onMouseLeave={[Function]}
    onMouseOver={[Function]}
  >
    <div
      className="place-card__mark"
    >
      <span>
        Premium
      </span>
    </div>
    <div
      className="cities__image-wrapper place-card__image-wrapper"
    >
      <a
        href="#"
      >
        <img
          alt="Place image"
          className="place-card__image"
          height="200"
          src="img/apartment-01.jpg"
          width="260"
        />
      </a>
    </div>
    <div
      className=" place-card__info"
    >
      <div
        className="place-card__price-wrapper"
      >
        <div
          className="place-card__price"
        >
          <b
            className="place-card__price-value"
          >
            €
            400
          </b>
          <span
            className="place-card__price-text"
          >
            / night
          </span>
        </div>
        <button
          className=" place-card__bookmark-button button"
          onClick={[Function]}
          type="button"
        >
          <svg
            className="place-card__bookmark-icon"
            height="19"
            width="18"
          >
            <use
              xlinkHref="#icon-bookmark"
            />
          </svg>
          <span
            className="visually-hidden"
          >
            To bookmarks
          </span>
        </button>
      </div>
      <div
        className="place-card__rating rating"
      >
        <div
          className="place-card__stars rating__stars"
        >
          <span
            style={
              Object {
                "width": "100%",
              }
            }
          />
          <span
            className="visually-hidden"
          >
            Rating
          </span>
        </div>
      </div>
      <h2
        className="place-card__name"
        onClick={[Function]}
      >
        <a
          href="/offer/7513497"
          onClick={[Function]}
        >
          Hilton palace
        </a>
      </h2>
      <p
        className="place-card__type"
      >
        hotel
      </p>
    </div>
  </article>
  <article
    className="cities__place-card place-card"
    onMouseLeave={[Function]}
    onMouseOver={[Function]}
  >
    <div
      className="place-card__mark"
    >
      <span>
        Premium
      </span>
    </div>
    <div
      className="cities__image-wrapper place-card__image-wrapper"
    >
      <a
        href="#"
      >
        <img
          alt="Place image"
          className="place-card__image"
          height="200"
          src="img/apartment-02.jpg"
          width="260"
        />
      </a>
    </div>
    <div
      className=" place-card__info"
    >
      <div
        className="place-card__price-wrapper"
      >
        <div
          className="place-card__price"
        >
          <b
            className="place-card__price-value"
          >
            €
            9
          </b>
          <span
            className="place-card__price-text"
          >
            / night
          </span>
        </div>
        <button
          className=" place-card__bookmark-button button"
          onClick={[Function]}
          type="button"
        >
          <svg
            className="place-card__bookmark-icon"
            height="19"
            width="18"
          >
            <use
              xlinkHref="#icon-bookmark"
            />
          </svg>
          <span
            className="visually-hidden"
          >
            To bookmarks
          </span>
        </button>
      </div>
      <div
        className="place-card__rating rating"
      >
        <div
          className="place-card__stars rating__stars"
        >
          <span
            style={
              Object {
                "width": "20%",
              }
            }
          />
          <span
            className="visually-hidden"
          >
            Rating
          </span>
        </div>
      </div>
      <h2
        className="place-card__name"
        onClick={[Function]}
      >
        <a
          href="/offer/9431675"
          onClick={[Function]}
        >
          Bungalo
        </a>
      </h2>
      <p
        className="place-card__type"
      >
        house
      </p>
    </div>
  </article>
</div>
`;
